### PR TITLE
Add Project Class, DateTime Class and minor edits

### PR DESCRIPTION
**Pull request by scimad**
Instead of creating a project directly from 'base' class, I have tried to create a separate 'Project' class that contains the details of the project to be created. The project detail is then injected to the constructor of the Base class and then does the rest.

I 'obviously' doubt if it's the best thing to do. So I would like you @pranphy to comment on it after reviewing the changes.

### CODE REVIEW

File: include/Base/Base.h [hunk 1]
```diff
@@ -1,8 +1,19 @@
...
 #include <map>
-class Base
-{
+class Project{
+public:
+	std::string ProjectName;
+	//std::string Path;
+	std::string Author;
+	std::string Email;
+	std::string TimeStamp;
```
Comment (pranphy) [inline include/Base/Base.h:10]: Make them private, and write getters and setters if you need them.

File: src/Base/Base.cpp [hunk 2]
```diff
@@ -37,40 +52,49 @@ void Base::CreateReadme(){
...
+	std::cout<<"Strlen = "<<StrLen<<std::endl;
+    bool Found = false;
+    std::string StartString, EndString, ParsedString = RawLine, TempKey; 
 	if (StrLen>4){
-		for (int i=0; i<StrLen-4; i++){
-				if (StartString == "{{") break;			
-		}
-		
+		for (i=0; i<StrLen-4; i++){
+            StartString = RawLine.substr(i,2);
```
Comment (pranphy) [inline src/Base/Base.cpp:69]: Configure editor to have four space tab. You may use astyle formatter in the beginning for consistency and then use proper editor.

**Review by pranphy — COMMENTED** [review]
ल पासा यत्ति गरौँ।

File: include/Base/Base.h (hunk 1 above)
Comment (scimad) [reply include/Base/Base.h:10]: I knew that drills but I wanted to make the project running and wrap up.

**Pull request merged**